src/test/fs_tests.cpp

```diff
@@ -11,6 +11,26 @@
 
 BOOST_FIXTURE_TEST_SUITE(fs_tests, BasicTestingSetup)
 
+BOOST_AUTO_TEST_CASE(fsbridge_pathtostring)
+{
+    std::string u8_str = "fs_tests_₿_🏃";
+    BOOST_CHECK_EQUAL(fs::PathToString(fs::PathFromString(u8_str)), u8_str);
```

Comment:
I wonder if it is worth adding a test for [.stem()](https://en.cppreference.com/w/cpp/filesystem/path/stem) method too. We use it in [dbwrapper.cpp](https://github.com/bitcoin/bitcoin/pull/22937/files#diff-44bef8fa93d168379ef331d9e11d333be06ce98a8f37ec65f8a987846b90a151R118). 

The test can be simply:

```cpp
BOOST_CHECK_EQUAL(fs::PathToString(fs::PathFromString(u8_str).stem()), u8_str);
```

or something like

```cpp
std::string test_filename = "fs_tests_₿_🏃.dat";
std::string expected_stem = "fs_tests_₿_🏃";
BOOST_CHECK_EQUAL(fs::PathToString(fs::PathFromString(u8_filename).stem()), expected_stem);
```

The reasoning behind this is to have a test that tests: string -> path -> modified path -> string. So we would be more sure that there is no std::filesystem implementation that behaves in a surprising way (I'm thinking about Android std::filesystem implementation that seems to be quite new.)

Anyway, it's just an idea from a category "maybe nice to have" and not "must have". Feel free to ignore.